packages/vscode-extension/src/webview/views/ActivateLicenseView.tsx

```diff
@@ -0,0 +1,111 @@
+
+export function ActivateLicenseView() {
+  const { project } = useProject();
+  const { closeModal } = useModal();
+  const { register, handleSubmit } = useForm();
+
+  const [isLoading, setIsLoading] = useState(false);
+  const [disableSubmit, setDisableSubmit] = useState(true);
+  const [activateDeviceResult, setActivateDeviceResult] = useState<ActivateDeviceResult | null>(
+    null
+  );
+
+  const inputRef = useRef<HTMLInputElement>(null);
+
+  const onSubmit: SubmitHandler<FieldValues> = (e, data) => {
+    setIsLoading(true);
+
+    const activationPromise = project.activateLicense(data?.target[0].value);
+
+    activationPromise.then((activationResult) => {
+      if (activationResult === ActivateDeviceResult.succeeded) {
+        closeModal();
+      } else {
+        setActivateDeviceResult(activationResult);
+      }
+      setIsLoading(false);
+    });
+  };
+
+  const onChange = () => {
+    const newInput = inputRef.current?.value;
+    if (newInput) {
+      const isValidKey = /^[0-9A-F]{4}(?:-[0-9A-F]{4}){7}$/.test(newInput);
+      setDisableSubmit(!isValidKey);
+      return;
+    }
+    setDisableSubmit(true);
+  };
+
+  return (
+    <form className="container" onSubmit={handleSubmit(onSubmit)}>
+      <div className="info-row">
+        {!activateDeviceResult && (
+          <div className="info-text">
+            You can find your license key on the Radon IDE customer portal (
+            <a href="https://portal.ide.swmansion.com/" target="_blank" rel="noopener noreferrer">
+              link
+            </a>
+            ) If you don't have a license, you can purchase it{" "}
+            <a href="https://ide.swmansion.com/pricing" target="_blank" rel="noopener noreferrer">
+              here
+            </a>
+            .
+          </div>
+        )}
+        {activateDeviceResult === ActivateDeviceResult.unableToVerify && (
+          <div className="error-text">
+            Unable to verify the key. Please ensure your license key is correct. Check this{" "}
+            <a
+              href="https://ide.swmansion.com/docs/guides/activation-manual"
+              target="_blank"
+              rel="noopener noreferrer">
+              link
+            </a>{" "}
+            for instructions.
+          </div>
+        )}
+        {activateDeviceResult === ActivateDeviceResult.notEnoughSeats && (
+          <div className="error-text">
+            Your organization does not have any available seats left, you can purchase more on the
+            Radon IDE customer portal (
+            <a href="https://portal.ide.swmansion.com/" target="_blank" rel="noopener noreferrer">
+              link
+            </a>
+            ).
+          </div>
+        )}
+        {activateDeviceResult === ActivateDeviceResult.connectionFailed && (
+          <div className="error-text">
+            We ware unable to register your device due to the connection failing.
```

Comment:
> Activating license key requires a network connection. Make sure you are connected and try activating the key again. In case of any issues please consult our [license activation instructions]